vendor/vendor-ios/src/main/kotlin/com/malinskiy/marathon/ios/device/LocalListSimulatorProvider.kt

```diff
@@ -47,11 +48,17 @@ class LocalListSimulatorProvider(override val coroutineContext: CoroutineContext
 
     override suspend fun start() = withContext(coroutineContext) {
         logger.info("starts providing ${simulators.count()} simulator devices")
-        val jobs = simulators.map {
-            async(context = coroutineContext) {
-                createDevice(it, RemoteSimulatorConnectionCounter.putAndGet(it.udid))?.let { connect(it) }
-            }
-        }.joinAll()
+        val marathonStartDuration = measureTimeMillis {
+            val jobs = simulators.map {
+                async(context = coroutineContext + CoroutineName("creator")) {
+                    createDevice(it, RemoteSimulatorConnectionCounter.putAndGet(it.udid))?.let { connect(it) }
+                }
+            }.also {
+                logger.debug("dispatched ${it.size} async jobs")
+            }.joinAll()
+        }
+        logger.debug("completed all jobs with ${devices.mappingCount()} stored devices")
+        logger.info("##teamcity[buildStatisticValue key='marathonStartDuration' value='$marathonStartDuration']")
```

Comment:
I highly doubt this should ever be exposed directly with this. I'd be fine to merge this as long as we rework this later with proper analytics